browser/src/react-redux/App/SignupPage.jsx

```diff
@@ -50,34 +50,38 @@ class SignupPage extends Component {
     )
   }
 
-  onSubmit = (e) => {
-    this.setState(initialState);
+  onSubmit = e => {
+    this.setState(initialState)
 
-    e.preventDefault();
+    e.preventDefault()
 
     const user = {
       email: this.state.email,
       password: this.state.password
-    };
+    }
 
-    axios.post('/api/user', user)
+    axios
+      .post('/api/user', user)
       .then(res => {
         //set current user with redux store.
-        this.props.setUser(res.data.user);
+        this.props.setUser(res.data.user)
       })
       .catch(err => {
         if (err.response.data.error.errors) {
-          const errors = err.response.data.error.errors;
-
-          errors.email ? this.setState({ emailError: errors.email.message }) : null
-          errors.password ? this.setState({ passwordError: errors.password.message }) : null
-
+          const errors = err.response.data.error.errors
+
+          errors.email
+            ? this.setState({ emailError: errors.email.message })
+            : null
+          errors.password
+            ? this.setState({ passwordError: errors.password.message })
+            : null
         } else {
           this.setState({
             generalError: err.response.data.error.message
-          });
+          })
         }
-      });
+      })
   }
   componentDidUpdate(prevProps) {
     if (this.props.user !== prevProps.user) {
```

Comment:
Let's go ahead just always redirect when `this.props.user` is populated. We'll eventually hide the login link when someone is logged in. If they do end up bookmarking it or visiting it directly, this new requirement will simply redirect them as intended.